src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -98,11 +100,8 @@ public static TripInstruction alignTravelerToTrip(TravelerPosition travelerPosit
                 : null;
         }
 
-        Coordinates legDestination = new Coordinates(travelerPosition.expectedLeg.to);
-        double distanceToDestination = getDistance(travelerPosition.currentPosition, legDestination);
-        if (distanceToDestination <= TRIP_INSTRUCTION_UPCOMING_RADIUS) {
-            // Assuming traveler is approaching the destination.
-            return new TripInstruction(distanceToDestination, travelerPosition.expectedLeg.to.name);
+        if (isApproachingDestination(travelerPosition)) {
+            return new TripInstruction(getDistanceToDestination(travelerPosition), travelerPosition.expectedLeg.to.name);
         }
 
         Step nextStep = snapToStep(travelerPosition);
```

Comment:
At line 108: Add a check that the current position is actually not in a segment of the next step.

Reply:
@binh-dam-ibigroup Changes in https://github.com/ibi-group/otp-middleware/pull/227/commits/1ef640b13dbe36dad3f071e96864874753e19713. A change in location of mere centimeters can make difference between getting an instruction or not! If not sure how well that is going to perform in the field.